bgpd/bgp_ecommunity.c

```diff
@@ -1026,3 +1026,144 @@ int ecommunity_fill_pbr_action(struct ecommunity_val *ecom_eval,
 		return -1;
 	return 0;
 }
+
+static unsigned int bgp_aggr_ecommunity_hash_key(void *data)
+{
+	struct ecommunity *ecommunity = data;
+
+	return jhash(ecommunity->val, ecommunity->size, 0xdcba4321);
+}
+
+static bool bgp_aggr_ecommunity_hash_cmp(const void *n1, const void *n2)
+{
+	const struct ecommunity *ecom1 = n1;
+	const struct ecommunity *ecom2 = n2;
+
+	if (ecom1 == NULL && ecom2 == NULL)
```

Comment:
how can ecom1 or ecom2 ever be null?

Reply:
Yes. It is not possible for com1 and com2 to both be NULL at the same time. I've removed this check.